rpcserver.go

```diff
@@ -7195,34 +7196,40 @@ func (r *rpcServer) UpdateChannelPolicy(ctx context.Context,
 	}
 
 	// By default, positive inbound fees are rejected.
-	if !r.cfg.AcceptPositiveInboundFees {
-		if req.InboundBaseFeeMsat > 0 {
+	if !r.cfg.AcceptPositiveInboundFees && req.InboundFee != nil {
+		if req.InboundFee.BaseFeeMsat > 0 {
 			return nil, fmt.Errorf("positive values for inbound "+
 				"base fee msat are not supported: %v",
-				req.InboundBaseFeeMsat)
+				req.InboundFee.BaseFeeMsat)
 		}
-		if req.InboundFeeRatePpm > 0 {
+		if req.InboundFee.FeeRatePpm > 0 {
 			return nil, fmt.Errorf("positive values for inbound "+
 				"fee rate ppm are not supported: %v",
-				req.InboundFeeRatePpm)
+				req.InboundFee.FeeRatePpm)
 		}
 	}
 
+	// In case inbound fees are not specified no inbound fees will be
+	// applied but the previous value will be retained.
```

Comment:
How is the previous value being retained?

Reply:
`updateEdge` has been modified to check the option value, only setting the value if `Some`.